### PR TITLE
feat: stored data in Realm database and added logged data option in navigation panel

### LINKED ISSUE #1321: realm database should be used to store sensor data
**Actual Behaviour**

Currently, the sensor data is stored directly in the external storage.

**Expected Behaviour**

The sensor data should be stored in Realm database and when the user the user clicks on Export data then the data should be fetched from database and written to the external storage in CSV format

**Steps to reproduce it**

Use Realm DataBase in LuxMeterDataFragment

**Would you like to work on the issue?**

Yes

### LINKED ISSUE #1322: data logger activity should be there to show all the recorded sensor data in one place
**Actual Behaviour**

There is no activity that shows loged data from different instruments in one place

**Expected Behaviour**

There should be one activity that shows a list of all the sensor data in form of items.

**Steps to reproduce it**

Modify the ShowLoggedData Activity and use it for this issue

**Screenshots of the issue**

Screenshot of app having similar functionality

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1322 #1316 
Fixes #1321 

**Changes**: 

- Stored the lux sensor data in Realm database in **LuxData.class** model instead of CSV file.
- Added lux data records in form of card items with **delete** option in newly created **DataLogger Activity**.
- Added Logged Data in navigation panel that opens DataLoggerActivity
- Prevented the unnecessary repetition of data by removing for loop
- Created option to open DataLoggerActivity from lux meter menu.**(As discussed in previous meeting)**

> To do: Show the data in Logged Data Activity in form of graph instead of the list.

**Screenshot/s for the changes**
Below GIF shows the complete workflow[Updated - 11/08/2018]

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: [realm.zip](https://github.com/fossasia/pslab-android/files/2279095/realm.zip)

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
@Avjeet in the recent meeting there was a format suggested. Can you please implement that?

**Comment by Avjeet** [comment]
@CloudyPadmal Actually all these issues are linked in some way so for proper resolution of all of them I had to make one PR,

**Comment by Avjeet** [comment]
@CloudyPadmal I have updated the screenshot

File: app/src/main/java/org/pslab/activity/LuxDataAdapter.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.pslab.models.SensorLogged;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+
+/**
+ * Created by Your name on 03-08-2018.
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/activity/LuxDataAdapter.java]: Shall we remove this?

File: app/src/main/java/org/pslab/activity/LuxDataAdapter.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ */
+class LuxDataAdapter extends RecyclerView.Adapter<LuxDataAdapter.ViewHolder> {
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/activity/LuxDataAdapter.java]: Is it possible to create a generic adapter for every instrument?

File: app/src/main/java/org/pslab/activity/LuxDataAdapter.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
+        final SensorLogged temp = results.get(position);
+        holder.sensor.setText(temp.getSensor());
+        Date date = new Date(temp.getDateTimeStamp());
+        holder.dateTime.setText(String.valueOf(sdf.format(date)));
+        holder.sensor.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                RealmResults<LuxData> results = realm.where(LuxData.class).equalTo("foreignKey",temp.getUniqueRef()).findAll();
+                Log.i("realmResult",String.valueOf(results.size()));
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/activity/LuxDataAdapter.java]: Once you have finalized this PR, please remove this..

File: app/src/main/java/org/pslab/fragment/LuxMeterFragmentData.java [hunk 4]
```diff
@@ -306,68 +314,68 @@ private void visualizeData() {
...
+//                                                    .setMessage(R.string.delete_warning)
+//                                                    .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+//                                                        @Override
+//                                                        public void onClick(DialogInterface dialogInterface, int i) {
+//                                                            lux_logger.deleteFile();
+//                                                        }
+//                                                    })
+//                                                    .setNegativeButton(R.string.cancel, null)
+//                                                    .create()
+//                                                    .show();
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/fragment/LuxMeterFragmentData.java]: I guess these comments are in WIP right?

**Comment by Avjeet** [comment]
@CloudyPadmal I have removed the CSv saving part, for now, I will implement it in Logged Data Activity later.

File: app/src/main/java/org/pslab/activity/SensorLoggerListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import org.pslab.models.SensorLogged;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import io.realm.Realm;
+import io.realm.RealmResults;
+
+/**
+ * Created by Your name on 03-08-2018.
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/activity/SensorLoggerListAdapter.java]: Either change this to Avjeet or delete this template block

File: app/src/main/java/org/pslab/activity/ShowLoggedData.java [hunk 6]
```diff
@@ -42,6 +42,7 @@
 
 /**
  * Created by viveksb007 on 12/8/17.
+ * deprecated
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/activity/ShowLoggedData.java]: If this file is no needed, we can delete it right?

File: app/src/main/java/org/pslab/models/LuxData.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package org.pslab.models;
+
+
+import io.realm.RealmObject;
+
+
+/**
+ * Created by Your name on 31-07-2018.
```
Comment (CloudyPadmal) [inline app/src/main/java/org/pslab/models/LuxData.java]: Same here, template

File: app/src/main/res/layout/logger_data_item.xml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            android:layout_weight="0.6"
+            android:gravity="center_vertical"
+            android:orientation="vertical">
+
+            <TextView
+                android:id="@+id/sensor_name"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="5dp"
+                android:text="Lux Meter"
```
Comment (CloudyPadmal) [inline app/src/main/res/layout/logger_data_item.xml]: I guess these are dummy texts right?

**Comment by Avjeet** [comment]
@CloudyPadmal @harsh-2711 @abhinavraj23 I have updated this PR as per discussed in meeting and uploaded the apk file, plz review it asap.
I will make the changes regarding setting part in next PR.

**Comment by cweitat** [comment]
@Avjeet  conflicts pls solve

**Comment by Avjeet** [comment]
@CloudyPadmal @harsh-2711 @abhinavraj23  I have rebased the code with new icons pls review this

**Comment by Avjeet** [comment]
merged with #1347

**Pull request closed without merging**